curp/src/client/mod.rs

```diff
@@ -121,13 +157,24 @@ pub struct ClientBuilder<P: Protocol> {
     config: ClientConfig,
 }
 
-impl<P: Protocol> ClientBuilder<P> {
+/// A client builder with bypass with local server
+#[derive(Debug, Clone)]
+#[allow(clippy::module_name_repetitions)] // same as above
+pub struct ClientBuilderWithBypass<P: Protocol> {
+    /// inner builder
+    inner: ClientBuilder,
+    /// local server id
+    local_server_id: ServerId,
+    /// local server
+    local_server: P,
+}
+
+impl ClientBuilder {
     /// Create a client builder
     #[inline]
     #[must_use]
     pub fn new(config: ClientConfig) -> Self {
         Self {
-            local_server: None,
             cluster_version: None,
             all_members: None,
             leader_state: None,
```

Comment:
We'd better implement Default trait for the builders.

Reply:
done